frontend/src/services/CsvService.ts

```diff
@@ -0,0 +1,86 @@
+import { difference, keys } from 'lodash'
+import Papa from 'papaparse'
+
+import {
+  getTemplateParameters,
+  Template,
+} from '~features/templates/TemplatesService'
+
+export interface CsvInfo {
+  csvFilename: string
+  numRows: number
+  firstRowParams: Record<string, string>
+}
+
+/**
+ * Ensure that the keywords found in the template is all in the csv headers and each row.
+ * TODO: We can verify compulsory parameters like UIN and UIN_Type
+ */
+export const validateCsv = async ({
+  file,
+  template,
+}: {
+  file: File
+  template: Template
+}): Promise<CsvInfo> => {
+  const csvFilename = file.name
+  const requiredParams = getTemplateParameters(template)
```

Comment:
Do we need to account for case sensitivity?

Reply:
Case sensitivity is covered. With `Papa.parse(file, { header: true ...`, each row will have its field keyed with the header. If the keys don't match the required params exactly, the row would fail validation in the `validateRow` function.